### PR TITLE
maximum_angle_of_rotation_during_gravitational_maneuver

### PR DESCRIPTION
**Pull request by Zulusssss**
Add law and tests.

### CODE REVIEW

File: symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+# Description
+## A gravitational maneuver is a purposeful change in the trajectory and flight speed of a spacecraft under the influence
+## of the gravitational fields of celestial bodies.
+## The maximum angle of rotation of the rocket during a gravitational maneuver depends on the first cosmic velocity of
+## the planet around which the maneuver is performed, and the speed of the rocket relative to this planet.
+
+## Law is: phi = arctg((v1 / v2)^2), where
+## phi - maximum angle of rotation during a gravitational maneuver,
```
Comment (blackyblack) [inline symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py:19]: Angle between what?

File: test/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver_test.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from collections import namedtuple
+from pytest import fixture, raises
+from symplyphysics import (assert_equal, units, Quantity, errors,)
+from symplyphysics.laws.astronomy import maximum_angle_of_rotation_during_gravitational_maneuver as angle_law
+
+# Description
+## The Earth's first cosmic velocity is 7.91 kilometers per second. Let the speed of the rocket be equal to the Earth's
+## second cosmic velocity of 11.2 kilometers per second. Then the maximum angle of rotation during a gravitational maneuver
+## is 0.463 radians.
+## https://ru.wikipedia.org/wiki/%D0%9A%D0%BE%D1%81%D0%BC%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%B0%D1%8F_%D1%81%D0%BA%D0%BE%D1%80%D0%BE%D1%81%D1%82%D1%8C
```
Comment (blackyblack) [inline test/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver_test.py:10]: This link does not have any numbers related to your test.

Comment (Zulusssss) [reply test/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver_test.py:10]: > This link does not have any numbers related to your test.

There are the first and second cosmic velocities for the Earth. I did not find a specific tusk about the angle of rotation during a gravitational maneuver.

Comment (blackyblack) [reply test/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver_test.py:10]: Remove this link as it's misleading.

**Review by blackyblack — APPROVED** [review]

File: symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    angle_type,
+)
+
+# Description
+## A gravitational maneuver is a purposeful change in the trajectory and flight speed of a spacecraft under the influence
+## of the gravitational fields of celestial bodies.
+## The maximum angle of rotation of the rocket during a gravitational maneuver depends on the first cosmic velocity of
+## the planet around which the maneuver is performed, and the speed of the rocket relative to this planet.
+
+## Law is: phi = arctg((v1 / v2)^2), where
```
Comment (blackyblack) [inline symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py:18]: Btw, do you have a link to this law?

Comment (Zulusssss) [reply symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py:18]: > Btw, do you have a link to this law?

https://college.ru/astronomy/course/content/chapter3/section2/paragraph3/theory.html
image

File: symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    validate_input,
+    validate_output,
+    angle_type,
+)
+
+# Description
+## A gravitational maneuver is a purposeful change in the trajectory and flight speed of a spacecraft under the influence
+## of the gravitational fields of celestial bodies.
+## The maximum angle of rotation of the rocket during a gravitational maneuver depends on the first cosmic velocity of
+## the planet around which the maneuver is performed, and the speed of the rocket relative to this planet.
```
Comment (blackyblack) [inline symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py:16]: From your description someone may think that there are various angles during gravitiational maneuver. And you should choose any until it reaches maximum.
But actually this law describes the closest possible maneuver, when aiming range is at planet radius. Add it do description and to conditions.

**Review by blackyblack — APPROVED** [review]

File: symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+## Law describes the closest possible maneuver, when aiming range is at planet radius.
+
+## Law is: phi = arctg((v1 / v2)^2), where
+## phi - maximum angle of rotation during a gravitational maneuver (angle at which the velocity vector of the rocket rotates),
+## v1 - first cosmic velocity for a given planet,
+## v2 - rocket's velocity relative to the planet.
+
+# Conditions:
+# - law describes the closest possible maneuver, when aiming range is at planet radius.
```
Comment (blackyblack) [inline symplyphysics/laws/astronomy/maximum_angle_of_rotation_during_gravitational_maneuver.py:25]:
```suggestion
# - Aiming range equals to planet radius.
```

**Review by blackyblack — APPROVED** [review]

**Pull request merged**